src/lib.rs

```diff
@@ -25,12 +25,19 @@
 //! when compiled with the following features:
 //!
 //! ### `tokio`
-//! If you enable the `tokio` feature you can use [`Promise::spawn_async`] and [`Promise::spawn_blocking`]
+//! If you enable the `tokio` feature you can use [`Promise::spawn_async`], [`Promise::spawn_local`] and [`Promise::spawn_blocking`]
 //! which will spawn tasks in the surrounding tokio runtime.
 //!
 //! ### `web`
-//! If you enable the `web` feature you can use [`Promise::spawn_async`] which will spawn tasks using
+//! If you enable the `web` feature you can use [`Promise::spawn_local`] which will spawn tasks using
 //! [`wasm_bindgen_futures::spawn_local`](https://rustwasm.github.io/wasm-bindgen/api/wasm_bindgen_futures/fn.spawn_local.html).
+//!
+//! ### `smol`
+//! If you enable the `smol` feature you can use [`Promise::spawn_async`] and [`Promise::spawn_local`]
+//! which will spawn tasks using the smol executor. Remember to tick the smol executor with [`tick`] and [`tick_local`].
+//!
+//! ### tick-poll
+//! Enabling the `tick-poll` with `smol` calling [`Promise::poll`] will automatically tick the smol executor.
```

Comment:
Maybe write a bit about what the benefits and drawbacks of this is.

And also, I think it would be good to name the feature with the `smol` prefix, e.g. `smol_tick_poll`